include/rcutils/filesystem.h

```diff
@@ -114,7 +114,7 @@ rcutils_is_readable_and_writable(const char * abs_path);
  *         NULL on failure
  */
 RCUTILS_PUBLIC
-const char *
+char *
```

Comment:
The docblock should mention that the caller gets ownership of the allocated memory and how to free it.

Same for the internally used function `rcutils_concat`.

Reply:
Good call, will change that.

Reply:
I've updated the documentation for `rcutils_concat` and `rcutils_join_path` as requested.  Let me know if the style I did it in is consistent with what we are doing elsewhere in the codebase (I tried to make it do that, but there are some different styles around).